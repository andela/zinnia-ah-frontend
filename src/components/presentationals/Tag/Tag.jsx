import React from 'react';
import PropTypes from 'prop-types';

// styles
import './Tag.scss';

const Tag = ({ value, className, onClick }) => {
  return (
    <div
      className={`tag ${className}`}
      onClick={onClick}
      data-test="single-tag"
    >
      {value}
    </div>
  );
};

Tag.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string,
  onClick: PropTypes.func,
};

export default Tag;
