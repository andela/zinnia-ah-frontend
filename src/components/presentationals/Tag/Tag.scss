@import '../../../assets/variables';

.tags {
  border-radius: .3125rem;
  display: flex;
  font-size: .75rem;
  height: 1.4375rem;
  justify-content: flex-start;
  margin-bottom: 1rem;
  margin-right: 1rem;
  padding: 0 .625rem;
  width: inherit;
}

.tag {
  background: $secondary-grey;
  font-size: 1rem;
  height: max-content;
  margin: .5rem;
  padding: .5rem;
  width: max-content;

  &:hover {
    background: $primary-white;
    border: 1px solid $secondary-black;
    cursor: pointer;
  }
}
