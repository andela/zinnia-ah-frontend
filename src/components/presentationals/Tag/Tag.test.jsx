import React from 'react';
import { shallow } from 'enzyme';

import Tag from './Tag';

const props = {
  value: 'News',
  className: 'tag-lg',
};

describe('Article', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<Tag {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
