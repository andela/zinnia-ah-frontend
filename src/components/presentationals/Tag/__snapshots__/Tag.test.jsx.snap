// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article should render without errors 1`] = `
<div
  className="tag tag-lg"
  data-test="single-tag"
>
  News
</div>
`;
